src/main/java/org/jenkinsci/plugins/durabletask/WindowsBatchScript.java

```diff
@@ -59,34 +86,79 @@ public String getScript() {
         if (launcher.isUnix()) {
             throw new IOException("Batch scripts can only be run on Windows nodes");
         }
+
         BatchController c = new BatchController(ws);
 
-        String cmd;
+        List<String> launcherCmd = null;
+        FilePath binary;
+        if (FORCE_BINARY_WRAPPER && (binary = requestBinary(ws, c)) != null) {
+            launcherCmd = binaryLauncherCmd(c, ws, binary.getRemote(), c.getBatchFile2(ws).getRemote());
+            c.getBatchFile2(ws).write(script, "UTF-8");
+        }
+        if (launcherCmd == null) {
+            launcherCmd = scriptLauncherCmd(c, ws);
+        }
+
+        LOGGER.log(Level.FINE, "launching {0}", launcherCmd);
+        Launcher.ProcStarter ps = launcher.launch().cmds(launcherCmd).envs(escape(envVars)).pwd(ws).quiet(true);
+
+        /* Too noisy, and consumes a thread:
+        ps.stdout(listener);
+        */
+        ps.readStdout().readStderr(); // TODO see BourneShellScript
+        Proc p = ps.start();
+        c.registerForCleanup(p.getStdout());
+        c.registerForCleanup(p.getStderr());
+
+        return c;
+    }
+
+    @Nonnull
+    private List<String> binaryLauncherCmd(BatchController c, FilePath ws, String binaryPath, String scriptPath) throws IOException, InterruptedException {
+        String logFile = c.getLogFile(ws).getRemote();
+        String resultFile = c.getResultFile(ws).getRemote();
+        String outputFile = c.getOutputFile(ws).getRemote();
+        String controlDirPath = c.controlDir(ws).getRemote();
+
+        List<String> cmd = new ArrayList<>();
+        cmd.add(binaryPath);
+        cmd.add("-daemon");
+        cmd.add("-executable=cmd");
+        cmd.add("-args=/C call \\\"" + scriptPath + "\\\"");
```

Comment:
should this not be 2 commands?
```suggestion
        cmd.add("-args=/C");
        cmd.add("call \\\"" + scriptPath + "\\\"");
```

Reply:
In this case, the entire string is being supplied as the argument to the `-args` flag. If separated, the parsing would be off and the binary would not get the full string needed for the `-args` flag.